### PR TITLE
Kaitie/save rubric button functionality

### PR DESCRIPTION
**Pull request by kobryan0619**
This PR is intended to:

- Save the data from a newly created rubric to the rubric and learning goal model

This adds a component (`RubricEditor`) to hold all `LearningGoalItems` (see in red below).
image

In future work we will be updating the edit functionality of the rubric and learning goals as well as adding save/edit abilities for "Evidence Levels" (which are connected to a learning goal)

## Links

- [AITT 59](https://codedotorg.atlassian.net/jira/software/projects/AITT/boards/70?selectedIssue=AITT-59)

## Testing story

- In addition to the tests here, we will also want to have some eyes tests since state is being passed between so many components.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lib/levelbuilder/rubrics/LearningGoalItem.jsx [hunk 1]
```diff
@@ -5,11 +5,33 @@ import {borderRadius} from '@cdo/apps/lib/levelbuilder/constants';
...
+
+  const [learningGoalData] = useState(exisitingLearningGoalData);
 
   const handleCheckboxChange = () => {
-    setAiEnabled(!aiEnabled);
+    setUpdatedAiEnabled(!updatedAiEnabled);
+    handleAiEnabledChange(learningGoalData.id);
+  };
+
+  const handleKeyConceptChange = event => {
```
Comment (kobryan0619) [inline apps/src/lib/levelbuilder/rubrics/LearningGoalItem.jsx:22]: In order to pass data to and from all the components, I have added these handlers throughout the files.  Open to better/cleaner ways to accomplish this goal

File: apps/test/unit/lib/levelbuilder/rubrics/RubricsContainerTest.js [hunk 2]
```diff
@@ -10,29 +12,25 @@ describe('RubricsContainerTest', () => {
...
+    expect(wrapper.find('select#rubric_level_id option')).to.have.length(3);
+    expect(wrapper.find(LearningGoalItem)).to.have.length(1);
+    expect(wrapper.find('Button[text="Save your rubric"]')).to.have.length(1);
+  });
 
-    // validate that the item has been deleted
-    expect(wrapper.find('LearningGoalItem').length).to.equal(2);
-    expect(wrapper.find('LearningGoalItem').at(0).key()).to.equal('2');
+  it('adds a new learning goal on "Add new Key Concept" button click', () => {
+    const wrapper = mount(<RubricsContainer {...defaultProps} />);
```
Comment (kobryan0619) [inline apps/test/unit/lib/levelbuilder/rubrics/RubricsContainerTest.js:27]: I needed mount here because we are checking the props of some children components - I tried using `shallow` and it wasn't working

File: dashboard/app/controllers/rubrics_controller.rb [hunk 3]
```diff
@@ -1,10 +1,43 @@
...
+    @lesson = @rubric.lesson
+    if @rubric.save
+      redirect_to(edit_rubric_path(@rubric.id))
+      return @rubric.id
+    else
+      render :new
+    end
+  end
+
+  def update
```
Comment (kobryan0619) [inline dashboard/app/controllers/rubrics_controller.rb:33]: Note that update and edit are still WIP.

File: dashboard/config/routes.rb [hunk 4]
```diff
@@ -335,7 +335,14 @@
       end
     end
 
-    resources :rubrics, only: [:edit, :new]
+    resources :rubrics, only: [:edit, :new, :create, :update]
+
+    # resources :lessons, only: [:edit, :update] do
```
Comment (kobryan0619) [inline dashboard/config/routes.rb]: Note to self... I am just now seeing this comment in the routes file.  I'd like to see if it passes drone first and then go back to delete this.

File: dashboard/app/models/learning_goal.rb [hunk 5]
```diff
@@ -35,4 +35,20 @@ def generate_key
     return if key.present?
     self.key = SecureRandom.uuid
   end
+
+  def summarize
```
Comment (bethanyaconnor) [inline dashboard/app/models/learning_goal.rb]: Is this used anywhere? (asking because I think I created a function with the same name 😅 )

Comment (bethanyaconnor) [reply dashboard/app/models/learning_goal.rb]: oh it's used below in `summarize_for_rubric_edit`! Any reason not to combine these functions?

File: dashboard/config/routes.rb (hunk 4 above)
Comment (bethanyaconnor) [inline dashboard/config/routes.rb]: Leftover?

File: apps/src/lib/levelbuilder/rubrics/LearningGoalItem.jsx (hunk 1 above)
Comment (bethanyaconnor) [reply apps/src/lib/levelbuilder/rubrics/LearningGoalItem.jsx:22]: I'm curious if `updateAiEnabled` and `updatedKeyConcept` are needed or if you can just use the corresponding fields in  `existingLearningGoalData`

File: apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx [hunk 6]
```diff
@@ -1,49 +1,166 @@
...
+    event.preventDefault();
+
+    const newKey = generateLearningGoalKey();
+    const newId = newKey;
+    const startingData = {
+      key: newKey,
+      id: newId,
+      learningGoal: '',
+      aiEnabled: false,
+    };
```
Comment (bethanyaconnor) [inline apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx]: would it be possible to extract this into a function and also use it to instantiate the first learning goal?

File: apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx [hunk 7]
```diff
@@ -1,49 +1,166 @@
...
+    const dataUrl = !!rubric ? `/rubrics/${rubric.id}/edit` : RUBRIC_PATH;
+    const method = !!rubric ? 'PATCH' : 'POST';
+
+    // Checking that the csrf-token exists since it is disabled on test
+    const csrfToken = document.querySelector('meta[name="csrf-token"]')
+      ? document.querySelector('meta[name="csrf-token"]').attributes['content']
+          .value
+      : null;
+
+    const learningGoalListAsData = transformKeys(learningGoalList);
```
Comment (bethanyaconnor) [inline apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx:110]: Not sure where to put this comment and this may not be the right place to resolve but one thing that's missing in saving the learning goals is the `position` attribute (ie the order that the learning goals should appear). I think you could either set that attribute here, before sending the learning goals off to the server or on the server side -- up to you.

File: apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx [hunk 8]
```diff
@@ -1,49 +1,166 @@
...
+        let redirectUrl = response.url;
+        navigateToHref(redirectUrl);
+      })
+      .catch(err => {
+        console.error('Error saving rubric:' + err);
+      });
+  };
+
+  function transformKeys(startingList) {
+    let newList = [];
```
Comment (bethanyaconnor) [inline apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx]: nit: I think you can use `const` instead of `let` here

File: apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx [hunk 9]
```diff
@@ -1,49 +1,166 @@
...
+
+  const handleDropdownChange = event => {
+    setSelectedLevelForAssessment(event.target.value);
   };
 
   // TODO: In the future we might want to filter the levels in the dropdown for "submittable" levels
   //  "submittable" is in the properties of each level in the list.
   return (
-    <div>
+    <form id={FORM_ID}>
```
Comment (bethanyaconnor) [inline apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx]: Why is this needed? (this may also be the reason for all of the weird button behavior)

File: apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx [hunk 10]
```diff
@@ -52,8 +169,10 @@ export default function RubricsContainer({
       <div style={styles.containerStyle}>
         <label>Choose a level for this rubric to be evaluated on</label>
         <select
+          id={'rubric_level_id'}
```
Comment (bethanyaconnor) [inline apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx]: nit: the curly braces aren't needed here

File: apps/src/sites/studio/pages/rubrics/new.js [hunk 11]
```diff
@@ -8,12 +8,14 @@ $(document).ready(() => {
   const unitName = lessonData.unitName;
   const lessonNumber = lessonData.lessonNumber;
   const levels = lessonData.levels;
+  const lessonId = lessonData.id;
```
Comment (bethanyaconnor) [inline apps/src/sites/studio/pages/rubrics/new.js]: nit: you can do `const {unitName, lessonNumber, levels, lessonId} = lessonData` here to make this a bit more succinct

File: dashboard/app/controllers/rubrics_controller.rb [hunk 12]
```diff
@@ -1,10 +1,43 @@
...
+    @rubric = Rubric.find(params[:id])
+    @lesson = @rubric.lesson
+    # @rubric.update(rubric_params)
+    # @rubric.reload
```
Comment (bethanyaconnor) [inline dashboard/app/controllers/rubrics_controller.rb]: nit: remove commented out lines

File: dashboard/app/controllers/rubrics_controller.rb [hunk 13]
```diff
@@ -1,10 +1,43 @@
...
     @lesson = Lesson.find_by(id: params[:lessonId])
+    @rubric = Rubric.new
+  end
+
+  def create
+    @rubric = Rubric.new(rubric_params)
+    @lesson = @rubric.lesson
+    if @rubric.save
+      redirect_to(edit_rubric_path(@rubric.id))
+      return @rubric.id
```
Comment (bethanyaconnor) [inline dashboard/app/controllers/rubrics_controller.rb]: This behavior still confuses me. IMO, I think it would be more readable if you did something like
```
render json: {redirectUrl: edit_rubric_path(@rubric.id)}
```
then updated the javascript to match. Thoughts?

File: dashboard/app/controllers/rubrics_controller.rb (hunk 3 above)
Comment (bethanyaconnor) [reply dashboard/app/controllers/rubrics_controller.rb:33]: Would it mess you up a lot to remove `update` in this PR? If so, can you add a comment like `TODO(kaitie): fix and test update action`

File: dashboard/app/controllers/rubrics_controller.rb [hunk 14]
```diff
@@ -1,10 +1,43 @@
...
     @rubric = Rubric.find_by(id: params[:id])
     @lesson = @rubric.lesson
   end
 
   def new
     @lesson = Lesson.find_by(id: params[:lessonId])
+    @rubric = Rubric.new
+  end
+
+  def create
```
Comment (bethanyaconnor) [inline dashboard/app/controllers/rubrics_controller.rb:20]: When chatting with Tyrone, Mark, and Eric the other day, I realized I've gotten lazy on a convention to add the corresponding route above controller actions. Do you mind doing that in this controller? The [reference_guides_controller](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/controllers/reference_guides_controller.rb) does this really well as an example.

For example, I believe the route for the `create` action is `POST /rubrics`. For `new`, it's `GET /rubrics/new`. etc

Comment (bethanyaconnor) [inline dashboard/app/controllers/rubrics_controller.rb:20]: Can you add a test for `create` that checks if a Rubric and LearningGoals are created correctly?

File: dashboard/app/controllers/rubrics_controller.rb [hunk 15]
```diff
@@ -1,10 +1,43 @@
...
+    @rubric = Rubric.new(rubric_params)
+    @lesson = @rubric.lesson
+    if @rubric.save
```
Comment (bethanyaconnor) [inline dashboard/app/controllers/rubrics_controller.rb:23]: oh, one thing I should've mentioned -- you'll need to call `@rubric.lesson.unit.write_script_json` when the save succeeds. This is how we get content from levelbuilder to production (you'll need to do this on update as well when you get there). Happy to chat more about this!

File: dashboard/app/controllers/rubrics_controller.rb [hunk 16]
```diff
@@ -1,10 +1,43 @@
 class RubricsController < ApplicationController
+  include Rails.application.routes.url_helpers
```
Comment (bethanyaconnor) [inline dashboard/app/controllers/rubrics_controller.rb:2]: Can you add `before_action :require_levelbuilder_mode_or_test_env` here? (There are many examples of using this but [here's](https://github.com/code-dot-org/code-dot-org/blob/55677a769a54248329f56db1b8d6664737f93641/dashboard/app/controllers/data_docs_controller.rb#L2) one). This restricts these routes to only levelbuilder or test so that they can't be used in production.

**Review by bethanyaconnor — COMMENTED** [review]
Excellent work! I left some comments, a lot of which are things I forgot to mention before 😅 

One other thing that I probably should've noticed in a previous PR is restricting access to these actions. Can you test whether a non-levelbuilder can create rubrics? My guess is that there's a little bit of permissions work that'll be needed -- happy to pair with you tomorrow on that!

File: apps/src/lib/levelbuilder/rubrics/LearningGoalItem.jsx (hunk 1 above)
Comment (kobryan0619) [reply apps/src/lib/levelbuilder/rubrics/LearningGoalItem.jsx:22]: I can confirm, that this works!   I made the change.

File: dashboard/app/controllers/rubrics_controller.rb (hunk 3 above)
Comment (kobryan0619) [reply dashboard/app/controllers/rubrics_controller.rb:33]: I think it is SO close to working!  I actually got it working for a brief second but then broke something else so I undid it all.  In an effort to focus on just getting this PR merged I will leave the TODO comment as recommended.

File: dashboard/app/models/learning_goal.rb (hunk 5 above)
Comment (kobryan0619) [reply dashboard/app/models/learning_goal.rb]: I had relied on how activity_section.rb and lesson_activity.rb was set-up which is how this got its current form.  On second look, it seems like `summarize` was used more than once in each of those files, so, yes I agree on combining them.

File: apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx (hunk 6 above)
Comment (kobryan0619) [reply apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx]: I had some trouble with this because the function depends on knowing the length of this list before the list is created...  I am planning to chat about it tomorrow.

File: apps/src/lib/levelbuilder/rubrics/LearningGoalItem.jsx (hunk 1 above)
Comment (bethanyaconnor) [reply apps/src/lib/levelbuilder/rubrics/LearningGoalItem.jsx:22]: Sorry, I'm missing the change you made. Can you point to the commit?

File: apps/src/lib/levelbuilder/rubrics/RubricEditor.jsx [hunk 17]
```diff
@@ -0,0 +1,45 @@
...
+  addNewConcept,
+  deleteItem,
+  learningGoalList,
+  updateLearningGoal,
+}) {
+  const renderLearningGoalItems = learningGoalList.map(goal => {
+    return (
+      <LearningGoalItem
+        deleteItem={() => deleteItem(goal.id)}
+        id={goal.id}
```
Comment (bethanyaconnor) [inline apps/src/lib/levelbuilder/rubrics/RubricEditor.jsx]: nit: unneeded (and possibly not used by React)

File: apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx [hunk 18]
```diff
@@ -1,43 +1,153 @@
...
+
+  // TODO: Check that there is at least one programming level here
+  const initialLevelForAssessment = !!rubric ? rubric.levelId : levels[0].id;
+  const [selectedLevelForAssessment, setSelectedLevelForAssessment] = useState(
+    initialLevelForAssessment
+  );
+
+  const saveRubric = event => {
+    event.preventDefault();
+    const dataUrl = !!rubric ? `/rubrics/${rubric.id}/edit` : RUBRIC_PATH;
```
Comment (bethanyaconnor) [inline apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx]: I may have led you astray here at some point -- got some of the variables confused -- but the update path should be `/rubrics/${rubric.id}` (based on [this documentation](https://guides.rubyonrails.org/routing.html#crud-verbs-and-actions))

File: apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx [hunk 19]
```diff
@@ -1,43 +1,153 @@
...
+      method: method,
+      headers: {
+        'Content-Type': 'application/json',
+        'X-CSRF-Token': csrfToken,
+      },
+      body: JSON.stringify(rubric_data),
+    })
+      .then(response => response.json())
+      .then(data => {
+        navigateToHref(data.redirectUrl);
```
Comment (bethanyaconnor) [inline apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx:128]: just noting that you'll probably want to only do this when creating a rubric, rather than on update. This doesn't need to be resolved in this PR though

File: apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx [hunk 20]
```diff
@@ -1,43 +1,153 @@
...
+  };
+
+  function transformKeys(startingList) {
+    const newList = [];
 
-    // update the goal list
-    setLearningGoalList(oldList => {
-      return [...oldList, {id: updatedId}];
+    startingList.forEach(item => {
+      let newItem = {};
```
Comment (bethanyaconnor) [inline apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx]: nit: you can you `const` here

File: apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx [hunk 21]
```diff
@@ -1,43 +1,153 @@
...
+      .then(response => response.json())
+      .then(data => {
+        navigateToHref(data.redirectUrl);
+      })
+      .catch(err => {
+        console.error('Error saving rubric:' + err);
+      });
+  };
+
+  function transformKeys(startingList) {
```
Comment (bethanyaconnor) [inline apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx:138]: This function looks good but do you mind adding a comment explaining what it's doing?

File: apps/test/unit/lib/levelbuilder/rubrics/RubricEditorTest.js [hunk 22]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import {shallow} from 'enzyme';
+import LearningGoalItem from '@cdo/apps/lib/levelbuilder/rubrics/LearningGoalItem';
+import Button from '@cdo/apps/templates/Button';
+import {expect} from '../../../../util/reconfiguredChai';
+import RubricEditor from '@cdo/apps/lib/levelbuilder/rubrics/RubricEditor';
+import sinon from 'sinon';
+
+describe('RubricEditorTest ', () => {
+  let wrapper;
```
Comment (bethanyaconnor) [inline apps/test/unit/lib/levelbuilder/rubrics/RubricEditorTest.js:10]: curious why `wrapper` is global here?

Comment (bethanyaconnor) [reply apps/test/unit/lib/levelbuilder/rubrics/RubricEditorTest.js:10]: oh I see because it's created in the `beforeEach`

File: apps/test/unit/lib/levelbuilder/rubrics/RubricEditorTest.js [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+  ];
+
+  beforeEach(() => {
+    wrapper = shallow(
+      <RubricEditor
+        learningGoalList={sampleLearningGoalList}
+        addNewConcept={addNewConceptSpy}
+        deleteItem={deleteItemSpy}
+        handleAiEnabledChange={hndleAiEnabledChangeSpy}
+        handleLearningGoalNameChange={handleLearningGoalNameChangeSpy}
```
Comment (bethanyaconnor) [inline apps/test/unit/lib/levelbuilder/rubrics/RubricEditorTest.js]: nit: you can just stub out the function if you're not going to test it (ie `deletedItem={() => {}}`)

Comment (bethanyaconnor) [reply apps/test/unit/lib/levelbuilder/rubrics/RubricEditorTest.js]: If you want to keep them for testing in the future, my other nit is `hndleAiEnabledChangeSpy` should be `handleAiEnabledChangeSpy`

File: apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx (hunk 21 above)
Comment (bethanyaconnor) [reply apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx:138]: thanks!

File: dashboard/app/controllers/rubrics_controller.rb (hunk 14 above)
Comment (bethanyaconnor) [reply dashboard/app/controllers/rubrics_controller.rb:20]: Thanks for doing this!

File: dashboard/test/controllers/rubrics_controller_test.rb [hunk 24]
```diff
@@ -0,0 +1,45 @@
+require 'test_helper'
+
+class RubricsControllerTest < ActionController::TestCase
+  include Devise::Test::ControllerHelpers
+
+  setup do
+    @levelbuilder = create :levelbuilder
+    sign_in @levelbuilder
```
Comment (bethanyaconnor) [inline dashboard/test/controllers/rubrics_controller_test.rb]: nit: I don't think you need this in the `setup` block

File: dashboard/test/controllers/rubrics_controller_test.rb [hunk 25]
```diff
@@ -0,0 +1,45 @@
...
+        ]
+      }
+    end
+
+    puts @rubric.inspect
+
+    response_json = JSON.parse(response.body)
+    rubric_id = response_json['rubricId']
+    @rubric = Rubric.find_by(id: rubric_id)
+    @learning_goals = LearningGoal.where(rubric_id: @rubric.id)
```
Comment (bethanyaconnor) [inline dashboard/test/controllers/rubrics_controller_test.rb]: nit: you don't need the `@` here because you're not using these globally. So these can just be `rubric = ...`

File: dashboard/test/controllers/rubrics_controller_test.rb [hunk 26]
```diff
@@ -0,0 +1,45 @@
...
+        level_id: @level.id,
+        lesson_id: @lesson.id,
+        learning_goals_attributes: [
+          {learning_goal: 'learning goal example 1', ai_enabled: true, position: 1},
+          {learning_goal: 'learning goal example 2', ai_enabled: false, position: 2}
+        ]
+      }
+    end
+
+    puts @rubric.inspect
```
Comment (bethanyaconnor) [inline dashboard/test/controllers/rubrics_controller_test.rb]: nit: remove debug statement

**Review by bethanyaconnor — COMMENTED** [review]
Great progress! I added a couple nits and there are a couple of outstanding comments that might've gotten lost, but this is so close. Thanks for the perseverance here :)

File: apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx [hunk 27]
```diff
@@ -1,43 +1,156 @@
...
-      deleteItem={() => deleteKeyConcept(goal.id)}
-      key={goal.id}
-    />
-  ));
+    return {
+      key: newKey,
+      id: newId,
+      learningGoal: '',
+      aiEnabled: false,
+      position: nextPosition,
```
Comment (kobryan0619) [inline apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx:57]: Added position into the empty KeyConcept

File: apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx (hunk 7 above)
Comment (kobryan0619) [reply apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx:110]: I did add the `position` to the[ key concepts. ](https://github.com/code-dot-org/code-dot-org/pull/53432/files#r1312201215).  It is now saving in the table.

File: apps/src/lib/levelbuilder/rubrics/LearningGoalItem.jsx (hunk 1 above)
Comment (kobryan0619) [reply apps/src/lib/levelbuilder/rubrics/LearningGoalItem.jsx:22]: [This](https://github.com/code-dot-org/code-dot-org/pull/53432/commits/f3416c5b1e1f9fe2ecd8bc9a5a4eff3a366894f4) is the commit where I made the change.  I ended up creating a `updateLearningGoal` function that would modify a learning goal's data and then using that function inside the handler.

Comment (bethanyaconnor) [reply apps/src/lib/levelbuilder/rubrics/LearningGoalItem.jsx:22]: gotcha! Do you still need `updatedAiEnabled` or can you just use `exisitingLearningGoalData.aiEnabled` (same thing for key concept)

File: apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx (hunk 27 above)
Comment (bethanyaconnor) [reply apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx:57]: oh thank you! Sorry I missed this!

File: apps/test/unit/lib/levelbuilder/rubrics/RubricsContainerTest.js [hunk 28]
```diff
@@ -10,29 +12,25 @@ describe('RubricsContainerTest', () => {
...
+    const initialLearningGoalItems = wrapper.find('LearningGoalItem').length;
+    wrapper.find('RubricEditor').invoke('addNewConcept')({
+      preventDefault: sinon.spy(),
+    });
```
Comment (davidsbailey) [inline apps/test/unit/lib/levelbuilder/rubrics/RubricsContainerTest.js]: the recommended way to write mount tests is to use ui elements rather than component methods. can you please update this to do the UI actions the user would do? otherwise, this will be hard to move to react testing library which we hope to do soon

File: apps/src/lib/levelbuilder/rubrics/LearningGoalItem.jsx (hunk 1 above)
Comment (davidsbailey) [reply apps/src/lib/levelbuilder/rubrics/LearningGoalItem.jsx:22]: sorry if i missed this @kobryan0619 but was there a resolution here?

File: apps/src/lib/levelbuilder/rubrics/RubricEditor.jsx [hunk 29]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import LearningGoalItem from './LearningGoalItem';
+import Button from '@cdo/apps/templates/Button';
+
+export default function RubricEditor({
```
Comment (davidsbailey) [inline apps/src/lib/levelbuilder/rubrics/RubricEditor.jsx:6]: this looks like some new UI, would you mind please including a screenshot showing what this adds? this isn't strictly necessary on every PR if everyone knows what's going on, but I'm afraid I'm still ramping up on the UI side of this feature

File: apps/test/unit/lib/levelbuilder/rubrics/RubricsContainerTest.js (hunk 28 above)
Comment (kobryan0619) [reply apps/test/unit/lib/levelbuilder/rubrics/RubricsContainerTest.js]: Oh, this makes sense, updating now!

File: apps/src/lib/levelbuilder/rubrics/LearningGoalItem.jsx (hunk 1 above)
Comment (kobryan0619) [reply apps/src/lib/levelbuilder/rubrics/LearningGoalItem.jsx:22]: Thanks all.  I will be making this change to just rely on `exisitingLearningGoalData.aiEnabled` (and `exisitingLearningGoalData.learningGoal`) instead of duplicating it in state.  I had a misconception that these couldn't be passed through as props.

File: apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx (hunk 7 above)
Comment (bethanyaconnor) [reply apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx:110]: Just checking -- we can't rearrange or delete key concepts yet right?

**Review by bethanyaconnor — APPROVED** [review]
Thanks for your perseverance here! Looks good to me :)
